lib/rpcs_dpdk/rpcs_dpdk.h

```diff
@@ -29,4 +30,69 @@ neg_errno_h2rpc(int *retval)
         *retval = -errno_h2rpc(-*retval);
 }
 
+/**
+ * Macro to convert RPC bitmask to/from RTE.
+ *
+ * @param _from               Bitmask to convert from.
+ * @param _to                 Bitmask to convert to.
+ * @param _rte2rpc            Whether conversion is from RTE to RPC.
+ * @param _conv_map           Conversion map.
+ * @param _rc                 Return code.
+ */
+#define RPC_DPDK_BITMASK64_CONVERT(_from, _to, _rte2rpc, _conv_map, _rc)      \
+    do {                                                                      \
+            uint64_t _bm = (_from);                                           \
+                                                                              \
+            (_rc) = te_enum_bitmask_convert((_conv_map), _bm,                 \
+                                            (_rte2rpc), &_bm);                \
+            if ((_rc) == 0)                                                   \
+                (_to) = _bm;                                                  \
+    } while (0)
+
+
+/**
+ * Macro to convert bitmask from RTE to RPC.
+ *
+ * @param _rte                RTE bitmask.
+ * @param _rpc                RPC bitmask.
+ * @param _bits_num           Number of bits in bitmask.
+ * @param _conv_map           Conversion map.
+ * @param _unknown_bit_name   Name of the bit to set if the conversion failed.
+ */
+#define RPC_DPDK_BITMASK_RTE2RPC(_rte, _rpc, _bits_num, _conv_map,            \
+                                 _unknown_bit_name)                           \
+    do {                                                                      \
+        unsigned int _unknown_bit =                                           \
+            TARPC_RTE_##_unknown_bit_name##__UNKNOWN_BIT;                     \
+        uint64_t _result;                                                     \
+        te_errno _rc;                                                         \
+                                                                              \
+        RPC_DPDK_BITMASK64_CONVERT((_rte), (_result), TRUE, (_conv_map), _rc);\
+        if (_rc == 0 && (_result == (_result & UINT##_bits_num##_C(-1))))     \
+            (_rpc) = (uint##_bits_num##_t)_result;                            \
+        else                                                                  \
+            (_rpc) = (UINT##_bits_num##_C(1) << _unknown_bit);                \
+    } while (0)
+
+/**
+ * Macro to convert bitmask from RPC to RTE.
+ *
+ * @param _rte                RTE bitmask.
+ * @param _rpc                RPC bitmask.
+ * @param _bits_num           Number of bits in bitmask.
+ * @param _conv_map           Convertsion map.
+ * @param _rc                 Return code.
+ */
+#define RPC_DPDK_BITMASK_RPC2RTE(_rpc, _rte, _bits_num, _rte_type,            \
```

Comment:
IMHO, two pair of functions for 32 and 64 bits will make the code better. Such macros are evil.